Create a emploi_store library to abstract the REST API with a nicer Python interface.

pcorpet opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/bayesimpact/python-emploi-store/1) &emsp; Multiple assignees:&emsp;<img alt="@dedan" height="20" width="20" align="absmiddle" src="https://avatars.githubusercontent.com/u/196905?s=40&v=3">&nbsp;<a href="/bayesimpact/python-emploi-store/pulls/assigned/dedan">dedan</a>,&emsp;<img alt="@florianjourda" height="20" width="20" align="absmiddle" src="https://avatars.githubusercontent.com/u/4361853?s=40&v=3">&nbsp;<a href="/bayesimpact/python-emploi-store/pulls/assigned/florianjourda">florianjourda</a>
<!-- Reviewable:end -->

pcorpet commented:
+@dedan +@florianjourda 

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-:-KEk4MzAngvLSX1KGuJE:793040211)*
<!-- Sent from Reviewable.io -->

dedan commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

And again I learned a lot from reading your code. Left a few comments, but ready to go..

---

Reviewed 3 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[emploi_store.py, line 25 \[r2\]](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-KEm4Y5SFdvYDMAM9nq-:-KEm4Y5SFdvYDMAM9nq0:-636863334) ([raw file](https://github.com/bayesimpact/python-emploi-store/blob/44498005acb5eba552795f680f2dd9f8ab6c4802/emploi_store.py#L25)):*
I would wonder, does the actual downloading happen here or in the `to_csv` line. Because the comment in the latter uses the word 'download'

---

*[emploi_store.py, line 63 \[r2\]](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-KEm5BGDg6b3KVSdgi2f:-KEm5BGDg6b3KVSdgi2g:1642770803) ([raw file](https://github.com/bayesimpact/python-emploi-store/blob/44498005acb5eba552795f680f2dd9f8ab6c4802/emploi_store.py#L63)):*
why do you have to generate a new access token every five seconds?

I did not ask this as a rhetorical question with the intention to make you change something. I have trouble understanding the logic of this piece of code

---

*[emploi_store.py, line 66 \[r2\]](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-KEm5YkfUsvUJMb-NpVG:-KEm5YkfUsvUJMb-NpVH:1814405222) ([raw file](https://github.com/bayesimpact/python-emploi-store/blob/44498005acb5eba552795f680f2dd9f8ab6c4802/emploi_store.py#L66)):*
do you return the existing token when it is expired? `next_request_time > self._token_expired` sounds like checking for expiration to me

---

*[emploi_store.py, line 117 \[r2\]](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-KEm6FDqSI5CBDS3MM_X:-KEm6FDrt_0YrbNvdbg2:-866745448) ([raw file](https://github.com/bayesimpact/python-emploi-store/blob/44498005acb5eba552795f680f2dd9f8ab6c4802/emploi_store.py#L117)):*
Why don't you use an `or` here?

---

*[emploi_store.py, line 126 \[r2\]](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-KEmKN6JcLEvnF_aG06u:-KEmKN6JcLEvnF_aG06v:-799871351) ([raw file](https://github.com/bayesimpact/python-emploi-store/blob/44498005acb5eba552795f680f2dd9f8ab6c4802/emploi_store.py#L126)):*
Would be good to give a bit more information about the concepts `package` and `resource`. I don't know what a package in the store means.

---

*[example.py, line 47 \[r2\]](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-KEmNRr0uaw5c2UsXhP2:-KEmNRr1cM0iunLqvF4p:-331224295) ([raw file](https://github.com/bayesimpact/python-emploi-store/blob/44498005acb5eba552795f680f2dd9f8ab6c4802/example.py#L47)):*
I get 'permission denied' when trying to bind to 80 on my machine. Works in the 3000 range though. Is that a mac thing?

---

*[README.md, line 7 \[r2\]](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-KEm46-Gm1DHntL489jV:-KEm46-HhgcGBIqFIAjR:723370892) ([raw file](https://github.com/bayesimpact/python-emploi-store/blob/44498005acb5eba552795f680f2dd9f8ab6c4802/README.md#L7)):*
Maybe at a sentence: "Usage information and setting of environment variables can be fund in the docstring"

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-:-KEmOd-Oic8QoKzUTt-Q:-1845310713)*
<!-- Sent from Reviewable.io -->

pcorpet commented:
Review status: 1 of 4 files reviewed at latest revision, 7 unresolved discussions.

---

*[emploi_store.py, line 25 \[r2\]](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-KEm4Y5SFdvYDMAM9nq-:-KEmXfPH6ZnxooSYXCZR:1331273314) ([raw file](https://github.com/bayesimpact/python-emploi-store/blob/44498005acb5eba552795f680f2dd9f8ab6c4802/emploi_store.py#L25)):*
I updated this comment: it's only the reference to the full file.

---

*[emploi_store.py, line 63 \[r2\]](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-KEm5BGDg6b3KVSdgi2f:-KEmXjylEF6BNcq9VQim:902765202) ([raw file](https://github.com/bayesimpact/python-emploi-store/blob/44498005acb5eba552795f680f2dd9f8ab6c4802/emploi_store.py#L63)):*
I'm asking for an access token that is valid for the next 5 seconds, I'm not saying that it should be new: if the previous one is still valid use it otherwise generate a new one.

---

*[emploi_store.py, line 66 \[r2\]](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-KEm5YkfUsvUJMb-NpVG:-KEm_YcV2EXNSXrrqBEe:-1055807556) ([raw file](https://github.com/bayesimpact/python-emploi-store/blob/44498005acb5eba552795f680f2dd9f8ab6c4802/emploi_store.py#L66)):*
Oooops, reversed logic, I should return it only if it's not expired yet. Thanks for catching.

---

*[emploi_store.py, line 117 \[r2\]](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-KEm6FDqSI5CBDS3MM_X:-KEmbHRKAbBlqDNBGmac:897951312) ([raw file](https://github.com/bayesimpact/python-emploi-store/blob/44498005acb5eba552795f680f2dd9f8ab6c4802/emploi_store.py#L117)):*
I don't see how I could: note that if package_id is None but name isn't I do something specific.

---

*[emploi_store.py, line 126 \[r2\]](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-KEmKN6JcLEvnF_aG06u:-KEmbO4v3yezec3u4ZaZ:-499146009) ([raw file](https://github.com/bayesimpact/python-emploi-store/blob/44498005acb5eba552795f680f2dd9f8ab6c4802/emploi_store.py#L126)):*
Done.

---

*[example.py, line 47 \[r2\]](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-KEmNRr0uaw5c2UsXhP2:-KEm_kBsr38IhN9By3ux:424517823) ([raw file](https://github.com/bayesimpact/python-emploi-store/blob/44498005acb5eba552795f680f2dd9f8ab6c4802/example.py#L47)):*
No it's a non-docker thing. By default I like my servers to run on port 80. If you want to change it because you're not an admin on your machine run

`PORT=3000 python example.py`

I added the documentation up there.

---

*[README.md, line 7 \[r2\]](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1#-KEm46-Gm1DHntL489jV:-KEmc86wTMRIArlpO6vO:-1834867264) ([raw file](https://github.com/bayesimpact/python-emploi-store/blob/44498005acb5eba552795f680f2dd9f8ab6c4802/README.md#L7)):*
Thanks.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bayesimpact/python-emploi-store/1)*
<!-- Sent from Reviewable.io -->
